jdk-wrapper.sh

```diff
@@ -189,6 +189,10 @@ extension="tar.gz"
 if [ "${JDKW_PLATFORM}" = "macosx-x64" ]; then
   extension="dmg"
 fi
+# JDK 6 and before use .bin extension, note no specific macosx download available for these versions (using linux)
+if [ "${JDKW_VERSION:2:1}" -le "6" ]; then
```

Comment:
This doesn't work.

I think you're looking for the prefix pattern match to digits. For example, if the Java version were ever 60u110 you wouldn't want to treat it as version 6. So we need to either extract the `60` (numeric prefix) or dump the suffix (starting at `u`).

Something like `${JDKW_VERSION%%u*}` should do it but I haven't test it thoroughly.